### PR TITLE
#57 EA-1986 If-Match check moved to commons-api 

### PR DESCRIPTION
**Pull request by rgraf**
for annotation- and entity-api

### CODE REVIEW

File: commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java [hunk 1]
```diff
@@ -163,6 +167,53 @@ private boolean isOperationAuthorized(String operation, List<GrantedAuthority> a
...
     }
 
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(String etag, HttpServletRequest request) throws HttpException {
```
Comment (gsergiu) [inline commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java:177]: this method must not be implemented in authorization service, it belogns to BaseRestControler

File: commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java [hunk 2]
```diff
@@ -163,6 +167,53 @@ private boolean isOperationAuthorized(String operation, List<GrantedAuthority> a
...
+    
+    /**
+     * This method generates etag for response header.
+     * 
+     * @param timestamp The date of the last modification
+     * @param format       The MIME format
+     * @param version      The API version
+     * @return etag value
+     */
+    public String generateETag(Date timestamp, String format, String version) {
```
Comment (gsergiu) [inline commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java:200]: this method must not be implemented in authorization service, it belogns to BaseRestControler

File: commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java [hunk 3]
```diff
@@ -38,4 +39,27 @@ public void verifyReadAccess(HttpServletRequest request)
...
+    
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(String etag, HttpServletRequest request) throws HttpException {
+    	getAuthorizationService().checkIfMatchHeader(etag, request);
```
Comment (gsergiu) [inline commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java:51]: the implementation must be here not delegated to authorizationService

File: commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java [hunk 4]
```diff
@@ -38,4 +39,27 @@ public void verifyReadAccess(HttpServletRequest request)
...
+    /**
+     * This method generates etag for response header.
+     * 
+     * @param timestamp The date of the last modification
+     * @param format       The MIME format
+     * @param version      The API version
+     * @return etag value
+     */
+    public String generateETag(Date timestamp, String format, String version) {
+		return getAuthorizationService().generateETag(timestamp, format, version);
```
Comment (gsergiu) [inline commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java:63]: the implementation must be here not delegated to authorizationService

File: pom.xml [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 	<groupId>eu.europeana.api.commons</groupId>
 	<artifactId>commons-api-services</artifactId>
-	<version>0.2.9-SNAPSHOT</version>
```
Comment (gsergiu) [inline pom.xml:9]: please merge the develop branch first to not overwrite the snapshot version in all modules

File: commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java (hunk 1 above)
Comment (rgraf) [reply commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java:177]: moved to BaseRestController

File: commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java (hunk 2 above)
Comment (rgraf) [reply commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java:200]: moved to BaseRestController

File: commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java (hunk 4 above)
Comment (rgraf) [reply commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java:63]: implemented in BaseRestController

File: pom.xml (hunk 5 above)
Comment (rgraf) [reply pom.xml:9]: Currently 2.8.0 used from develop

Comment (rgraf) [reply pom.xml:9]: I mean 0.2.8

**Pull request merged**